entity/entity_plugins/CMakeLists.txt

```diff
@@ -1,52 +1,28 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.5)
 project(entity_plugins)
 
-add_compile_options(-std=c++14)
-
-find_package(catkin REQUIRED COMPONENTS
-  pluginlib
-  scenario_actions
-  scenario_entities
-  scenario_logger
-  scenario_utility
-  )
-
-catkin_package(
-  INCLUDE_DIRS include
-  LIBRARIES ${PROJECT_NAME}
-  CATKIN_DEPENDS pluginlib
-                 scenario_actions
-                 scenario_entities
-                 scenario_logger
-                 scenario_utility
+### Compile options
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wno-unused-parameter -Wall -Wextra -Wpedantic)
+endif()
+
+find_package(ament_cmake_auto REQUIRED)
+# We require all dependencies but by default ament_auto_find_build_dependencies does not require packages
+ament_auto_find_build_dependencies(REQUIRED
+  ${${PROJECT_NAME}_BUILD_DEPENDS}
```

Comment:
Pretty slick. Where did you get these variables from :wink: ? Do you think we should open an issue with Dirk that REQUIRED should be the default?

Reply:
I am sceptical that this will be made the default, e.g. because some dependencies have different package.xml-names than CMake-names, but we can add this to the porting doc maybe.

Reply:
Can't hurt to try of course

Reply:
you are right, I had this issue with boost capitalization